src/main/main.py

```diff
@@ -43,39 +46,66 @@ def _get_mutation_cluster_annotation(request_data):
     except KeyError:
         return None
 
 
-@app.route("/mutate", methods=['POST'])
-def mutate():
+def _save_epsagon_instrumentation(deployment):
     """
-    Mutation admission main endpoint
-    Sends epsagon a notification about the change and
-    preserves the 'epsagon-mutatoin' label
+    Keeps epsagon changes on mutated deployment
     """
     epsagon_data = {'epsagon_token': TOKEN}
     epsagon_data.update(request.json)
-    deployment = request.json['request']['object']
-    modified_deployment = copy.deepcopy(deployment)
+    if 'labels' not in deployment['metadata']:
+        deployment['metadata']['labels'] = {}
 
-    if 'labels' not in modified_deployment['metadata']:
-        modified_deployment['metadata']['labels'] = {}
-
-    if "epsagon-auto-instrument" not in modified_deployment['metadata']['labels']:
+    if EPSAGON_AUTO_INST_FLAG not in deployment['metadata']['labels']:
         requests.post(app.config['EPSAGON_MUTATTIONS_ENDPOINT'], json=epsagon_data)
     else:
-        modified_deployment['metadata']['labels'].pop("epsagon-auto-instrument")
+        deployment['metadata']['labels'].pop(EPSAGON_AUTO_INST_FLAG)
 
-    modified_deployment['metadata']['labels']['epsagon-mutation'] = 'enabled'
+    deployment['metadata']['labels'][EPSAGON_MUTATION] = 'enabled'
     mutation_cluster = _get_mutation_cluster_annotation(request)
     if mutation_cluster:
-        if 'annotations' not in modified_deployment:
-            modified_deployment['metadata']['annotations'] = {}
-        modified_deployment['metadata']['annotations'][
-            'epsagon-mutation-cluster'] = mutation_cluster
-    patch = jsonpatch.JsonPatch.from_diff(deployment, modified_deployment)
+        if 'annotations' not in deployment:
+            deployment['metadata']['annotations'] = {}
+        deployment['metadata']['annotations'][
+            EPSAGON_MUTATION_CLUSTER] = mutation_cluster
+
+
+def _remove_epsagon_instrumentation(deployment):
+    """
+    Removes epsagon changes from mutated deployment
+    """
+    if "labels" in deployment['metadata']:
+        deployment['metadata']['labels'].pop(EPSAGON_REMOVE_AUTO_INST_FLAG, None)
+        deployment['metadata']['labels'].pop(EPSAGON_MUTATION, None)
+    if "annotations" in deployment['metadata']:
+        deployment['metadata']['annotations'].pop(EPSAGON_MUTATION_CLUSTER, None)
+
+
+@app.route("/mutate", methods=['POST'])
+def mutate():
+    """
+    Mutation admission main endpoint
+    Sends epsagon a notification about the change and
+    preserves the 'epsagon-mutatoin' label
+    """
+    try:
+        deployment = request.json['request']['object']
+        modified_deployment = copy.deepcopy(deployment)
+        if (
+                modified_deployment['metadata'].get("labels", {}).get(
+                    EPSAGON_REMOVE_AUTO_INST_FLAG, ""
+                ) == "disable"
+        ):
+            _remove_epsagon_instrumentation(modified_deployment)
+        else:
```

Comment:
I'd mark each operation individually, E.G: removal, modification, w/e just to avoid mistakes
u don't wanna fall into this else and modify when this is not actually a modification